tests/integration/buildcmd/test_build_terraform_applications.py

```diff
@@ -69,6 +68,91 @@ def run_command(self, command_list, env=None, input=None):
             process.kill()
             raise
 
+    def _verify_invoke_built_function(self, function_logical_id, overrides, expected_result):
+        LOG.info("Invoking built function '{}'".format(function_logical_id))
+
+        cmdlist = [
+            self.cmd,
+            "local",
+            "invoke",
+            function_logical_id,
+            "--no-event",
+            "--hook-package-id",
+            "terraform",
+            "--beta-features",
+        ]
+
+        if overrides:
+            cmdlist += [
+                "--parameter-overrides",
+                overrides,
+            ]
+
+        LOG.info("Running invoke Command: {}".format(cmdlist))
+
+        stdout, _, _ = self.run_command(cmdlist)
+
+        process_stdout = stdout.decode("utf-8")
+        self.assertEqual(json.loads(process_stdout), expected_result)
+
+
+class BuildTerraformApplicationS3BackendIntegBase(BuildTerraformApplicationIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        """Setting up a S3 bucket (using pre-created or create a new bucket) to use as a Terraform backend"""
+        cls.region_name = os.environ.get("AWS_DEFAULT_REGION", "us-east-1")
+
+        """Please read comments in package_integ_base.py for more details around this."""
+        bucket_env_var = os.environ.get("AWS_S3")
+        cls.pre_created_bucket = False
+        if bucket_env_var:
+            cls.pre_created_bucket = os.environ.get(bucket_env_var, False)
+        cls.bucket_name = cls.pre_created_bucket if cls.pre_created_bucket else str(uuid.uuid4())
+
+        s3 = boto3.resource("s3")
+        cls.s3_bucket = s3.Bucket(cls.bucket_name)
+        if not cls.pre_created_bucket:
+            cls.s3_bucket.create()
+            time.sleep(S3_SLEEP)
+
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean up and delete the bucket if it is not pre-created"""
+        cls.s3_bucket.objects.all().delete()
+        time.sleep(S3_SLEEP)
+        if not cls.pre_created_bucket:
+            cls.s3_bucket.delete()
+
+    def setUp(self):
+        self.backend_key = str(Path("terraform-backend") / str(uuid.uuid4()))
```

Comment:
does using '/' automatically join? if so should we use pathlib join operation instead as this may cause issues on windows.

Reply:
This should be fine. `Path("some/path")` will produce an instance of either `WindowsPath("some/path")` or `PosixPath("some/path"), depending on the OS. The `/` operator is just similar to pathlib join. 

On Windows:
```
>>> str(Path("some/path") / "child")
'some\\path\\child'
>>> Path("some/path") / "child"
WindowsPath('some/path/child')
>>>
```